### PR TITLE
fix(dns) allow resolution of service names with "."

### PR DESCRIPTION
**Pull request by lahabana**
When specifying a `kuma.io/service` with a "." DNS resolution would not work
because it would only look at the first part of the DNS name.
This would also cause `foo.bar.mesh` and `foo.mesh` to resolve to the same thing
which may give a false sense of isolation of meshes in the DNS
We now only strips the last part of the DNS name if it matches the `.mesh` domain suffix.

### CODE REVIEW

**Comment by lahabana** [comment]
This patch makes DNS resolution more restrictive so it's potentially not backward compatible depending on your treatment of undefined behavior as breaking changes.
Let me know if you'd prefer something backward compatible (I'm thinking of a flag in DNSServiceConfig).

File: pkg/dns/server_test.go [hunk 1]
```diff
@@ -121,5 +124,30 @@ var _ = Describe("DNS server", func() {
 			// and metrics are published
 			Expect(test_metrics.FindMetric(metrics, "dns_server_resolution", "result", "unresolved").Counter.GetValue()).To(Equal(1.0))
 		})
+
+		It("should resolve services with '.'", func() {
+			// given
+			dnsResolver.SetVIPs(vips.List{
+				"my.service": "240.0.0.1",
+			})
```
Comment (nickolaev) [inline pkg/dns/server_test.go:132]: If you do this in the `It` let's remove it from `BeforeEach` and put it in each `It`.

**Review by nickolaev — CHANGES_REQUESTED** [review]
Please fix the tests as suggested. Looks great otherwise

**Comment by lahabana** [comment]
@nickolaev updated thanks!

**Comment by lahabana** [comment]
@nickolaev seems like the build failure is unrelated can you retrigger the build?

**Review by nickolaev — APPROVED** [review]

**Pull request merged**